src/main/java/gregtech/api/recipes/machines/RecipeMapScanner.java

```diff
@@ -0,0 +1,49 @@
+package gregtech.api.recipes.machines;
+
+import gregtech.api.recipes.Recipe;
+import gregtech.api.recipes.RecipeMap;
+import gregtech.api.recipes.RecipeMaps;
+import gregtech.api.recipes.builders.SimpleRecipeBuilder;
+import gregtech.api.util.AssemblyLineManager;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraftforge.fluids.FluidStack;
+
+import javax.annotation.Nullable;
+import java.util.List;
+
+public class RecipeMapScanner extends RecipeMap<SimpleRecipeBuilder> implements IScannerRecipeMap {
+
+    public RecipeMapScanner(String unlocalizedName, int maxInputs, int maxOutputs, int maxFluidInputs, int maxFluidOutputs, SimpleRecipeBuilder defaultRecipe, boolean isHidden) {
+        super(unlocalizedName, maxInputs, maxOutputs, maxFluidInputs, maxFluidOutputs, defaultRecipe, isHidden);
+    }
+
+    @Override
+    @Nullable
+    public Recipe findRecipe(long voltage, List<ItemStack> inputs, List<FluidStack> fluidInputs, boolean exactVoltage) {
+        Recipe recipe = super.findRecipe(voltage, inputs, fluidInputs, exactVoltage);
+
+        // Data stick copying - min of 2 inputs required
+        if (recipe == null && inputs.size() > 1) {
+            ItemStack first = inputs.get(0);
+            ItemStack second = inputs.get(1);
+            NBTTagCompound compound = second.getTagCompound();
+            if (compound == null) return null;
+
```

Comment:
This will enforce the data sticks being placed into the slots in a specific configuration, which could be a bit annoying. It would be better if both the first and the second items were checked for the research key, that way the items could be placed into either slot. In cases where both items have the research key, take the key from the first item and override the key on the second item, and put it to user failure if that is not what they want.